GeneralsMD/Code/Main/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+add_executable(zerohour WIN32)
+
+target_link_directories(zerohour PRIVATE
+    ../GameEngine
+)
+
+target_link_libraries(zerohour PRIVATE
+    GameEngine
+    GameEngineDevice  
+    winmm
+    imm32
+    dbghelp
+    vfw32
+    d3d8
+    d3dx8
+    dinput8
+    dxguid
+    binkstub
+    milesstub
+    profile
+)
+
+# TODO Originally referred to build host and user, replace with git info perhaps?
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/GeneratedVersion.h
```

Comment:
This is what https://cmake.org/cmake/help/latest/command/configure_file.html#example was made for.

Reply:
Agreed, this was being quick and dirty to get a build up and running. Really what we want it something like https://github.com/TheAssemblyArmada/Vanilla-Conquer/blob/vanilla/common/gitinfo.cpp.in which we populate with appropriate information from git and maybe take version info from the project configuration if not on a git tag?

Reply:
(Opinion)
What I usually do is take a version from CI (git tag if present) and use that for the version, otherwise I use the latest tag before my commit and add a brief commit hash to the version. Something like this: https://stackoverflow.com/a/7261049